humanmade/hm-gutenberg-tools
# Add date range filters

This PR adds the ability to filter the posts returned by the PostSelect component based on selected dates. More details about context and technical brief can be found in the issue #89.

### Testing instructions

- Checkout the branch 89-add-date-range-filters
- Add an extra prop ```showDateFilters``` to the PostSelect component in your block
- Edit a post and add your block to the content so that the PostSelect component is shown
- Click the Select button to open the post select modal
- Two new buttons should show up in two columns titled "Published after date" and "Published date before" respectively
- Click the button below either of the columns to se the appropriate date
- Click "Filter posts" button at the bottom of modal screen and the posts returned should filter to include only those in the selected date range
- NB: if you set "Published after date" to say Thursday 4 November then posts published on the day will be included. Open to discussing if they shouldn't be included such that only posts truly published after that day would be returned.
- Also test that the "Published after date" cannot be after the "Published before date" and the reverse of the "Published before date" cannot be before the "Published after date" when that validation fails, an error notice should be shown.
